tags/TagMapContext.md

```diff
@@ -0,0 +1,144 @@
+- - Exceptions ':', ','
+- Must have length greater than zero and less than 256 combined with the scope.
+- Must not be empty.
+
+### Scope
+
+#### Scope Syntax
+
+```
+<scope-type1-code>,<scope-value1>,<scope-type2-code>,<scope-value2>,...
+```
+- Scope MUST always be in a Type-Value pair.
+
+There are following scopes types.
+
+#### Scope Types
+Scope Type is represented using code to reduce the size.
+
+| Code | Type  |
+|------|-------|
+| 0    | Process |
+| 1    | Internal-Service|
+| 2    | External-Service|
+
+
+##### Process
+A `TagKey` with the scope type *Process* is relevant only to stats collected by the process.
+Hence it should not be propagated across different processes. This is the default scope. If no
+scope is present in the `Tagkey` then it is treated as of Process scope type.
+
+##### Internal-Service
+A `TagKey` with the scope type *Internal-Service* is relevant to stats collected within an
+internal service which could be a single process or collection of process fulfilling the service.
+An internal services is the one that is not accessible from the public domain.
+An Example of internal service is a distributed caching service.
+
+A `TagKey` with this scope is propagated between different processes that makes up the internal
+service.
+
+
+##### External-Service
+A `TagKey` with *External Service* scope is stats collected across multiple 
+internal services that fulfills the external service which is accessible from another
+administrative domain.
+ 
+A `TagKey` with this scope is propagated between internal services.
+
+#### Example
+
+```
+TagKey1 = 2,RetailSvc:http_route
+TagValue1 = /cart/checkout
+
+TagKey2 = 1,CheckoutSvc:grpc_method
+TagValue2 = PlaceOrder
+
+TagKey3 = 0,ProductCatagalog:func_name
+TagValue3 = ParseCatalog
+
+```
+
+## TagValue
+
+`TagValue` is a string or string wrapper. It MUST contain only printable ASCII (codes between
+32 and 126)
+
+
+## Limits
+Combined size of all `TagKey`s and `TagValue`s should not exceed 8192 bytes before encoding.
+The size restriction applies to the deserialized tags so that the set of decoded
+ `TagMap`s is independent of the encoding format.
+
+
+# Propagation
+
+`Tag`s are propagated based on their scope. By default implementation should neither receive
+incoming `Tag`s nor it should forward any `Tag`s unless explicitly configured by the user.
+It MUST not receive or forward any `Tag` without the scope.
+
+## Recieving  
+**TBD**: how to sepcify receive criteria?
+
+## Forwarding
+**TBD**: how to specify forward criteria?
+
+## Encoding
+
+### Over gRPC
+TagMap should be encoded using [BinaryEncoding](https://github.com/census-instrumentation/opencensus-specs/tree/master/encodings)
+and propagated using gRPC metadata opencensus-tag-bin.
+
+### Over HTTP
+
+**TBD**
+                                                
+## Error handling
+
+- There are no partial failures.  The result of encoding or decoding
```

Comment:
error handling should not break callers. this section makes me think it is ok to raise error

Reply:
Comment#1 - Per tracer instance
I will clarify it.

Comment#2  - remote and not remote
Good point. Let me make the re-draft the spec.

Comment#3 - Error handling.
Let me separate out the encoding and error handling separately.
The intention here is not to break the call if there is error decoding the TagMap. But rather to accept or not accept partial data if there is an error. I'll clarify it.